### PR TITLE
Introduce custom ConfigMap for system-probe and move built-in configuration to ENV vars

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Currently the Operator uses config file instead of ENV vars, which means users cannot use a custom configuration file.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Deploy the Operator with:
- System probe enabled
- Using inline `CustomConfig` or using a custom ConfigMap name.

### CODE REVIEW

File: config/crd/bases/v1/datadoghq.com_datadogagents.yaml [hunk 1]
```diff
@@ -3464,6 +3464,30 @@ spec:
                           to connect to the netlink/conntrack subsystem to add NAT
                           information to connection data. See also: http://conntrack-tools.netfilter.org/'
                         type: boolean
+                      customConfig:
+                        description: Allow to put custom configuration for system-probe,
```
Comment (ruthnaebeck) [inline config/crd/bases/v1/datadoghq.com_datadogagents.yaml:3468]:
```suggestion
                        description: Enable custom configuration for system-probe,
```

File: config/crd/bases/v1/datadoghq.com_datadogagents.yaml [hunk 2]
```diff
@@ -3464,6 +3464,30 @@ spec:
...
+                          corresponding to the system-probe.yaml config file. This
+                          custom configuration has less priority than all settings
+                          above.
+                        properties:
+                          configData:
+                            description: ConfigData corresponds to the configuration
+                              file content.
+                            type: string
+                          configMap:
+                            description: ConfigMap name of a ConfigMap used to mount
```
Comment (ruthnaebeck) [inline config/crd/bases/v1/datadoghq.com_datadogagents.yaml:3478]:
```suggestion
                            description: The name of a ConfigMap used to mount
```

File: config/crd/bases/v1/datadoghq.com_datadogagents.yaml [hunk 3]
```diff
@@ -3464,6 +3464,30 @@ spec:
...
+                              the configuration file.
+                            properties:
+                              fileKey:
+                                description: FileKey corresponds to the key used in
+                                  the ConfigMap.Data to store the configuration file
+                                  content.
+                                type: string
+                              name:
+                                description: Name the ConfigMap name.
```
Comment (ruthnaebeck) [inline config/crd/bases/v1/datadoghq.com_datadogagents.yaml:3487]:
```suggestion
                                description: The name the ConfigMap name.
```

File: config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml [hunk 4]
```diff
@@ -3334,6 +3334,29 @@ spec:
                         to connect to the netlink/conntrack subsystem to add NAT information
                         to connection data. See also: http://conntrack-tools.netfilter.org/'
                       type: boolean
+                    customConfig:
+                      description: Allow to put custom configuration for system-probe,
```
Comment (ruthnaebeck) [inline config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml:3338]:
```suggestion
                      description: Enable custom configuration for system-probe,
```

File: config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml [hunk 5]
```diff
@@ -3334,6 +3334,29 @@ spec:
...
+                        corresponding to the system-probe.yaml config file. This custom
+                        configuration has less priority than all settings above.
+                      properties:
+                        configData:
+                          description: ConfigData corresponds to the configuration
+                            file content.
+                          type: string
+                        configMap:
+                          description: ConfigMap name of a ConfigMap used to mount
```
Comment (ruthnaebeck) [inline config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml:3347]:
```suggestion
                          description: The name of a ConfigMap used to mount
```

File: config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml [hunk 6]
```diff
@@ -3334,6 +3334,29 @@ spec:
...
+                            the configuration file.
+                          properties:
+                            fileKey:
+                              description: FileKey corresponds to the key used in
+                                the ConfigMap.Data to store the configuration file
+                                content.
+                              type: string
+                            name:
+                              description: Name the ConfigMap name.
```
Comment (ruthnaebeck) [inline config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml:3356]:
```suggestion
                              description: The name the ConfigMap name.
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Docs review

**Comment by vboulineau** [comment]
@ruthnaebeck I've integrated 2 of your suggestions and entirely rephrased the other one as the pre-existing doc phrase was not clear enough. Let me know if it's good to go.

File: config/manager/kustomization.yaml [hunk 7]
```diff
@@ -2,7 +2,7 @@ resources:
 - manager.yaml
 images:
 - name: controller
-  newName: gcr.io/datadoghq/operator
-  newTag: 0.6.0
+  newName: vboulineau/operator
```
Comment (lebauce) [inline config/manager/kustomization.yaml:5]: Is it on purpose ?

Comment (clamoriniere) [reply config/manager/kustomization.yaml:5]: don't think so :)

File: controllers/datadogagent/agent_test.go [hunk 8]
```diff
@@ -1050,6 +1074,7 @@ func defaultSystemProbePodSpec(dda *datadoghqv1alpha1.DatadogAgent) corev1.PodSp
 			SubPath:   "system-probe.yaml",
 		},
 	}...)
+	agentEnvVars := addEnvVar(defaultEnvVars(nil), datadoghqv1alpha1.DDSystemProbeSocketPath, filepath.Join(datadoghqv1alpha1.SystemProbeSocketVolumePath, "sysprobe.sock"))
```
Comment (clamoriniere) [inline controllers/datadogagent/agent_test.go:1077]: nit: add a empty line before `return`
```suggestion
	agentEnvVars := addEnvVar(defaultEnvVars(nil), datadoghqv1alpha1.DDSystemProbeSocketPath, filepath.Join(datadoghqv1alpha1.SystemProbeSocketVolumePath, "sysprobe.sock"))

```

File: controllers/datadogagent/controller_test.go [hunk 9]
```diff
@@ -736,7 +736,7 @@ func TestReconcileDatadogAgent_Reconcile(t *testing.T) {
 			wantErr: false,
 			wantFunc: func(c client.Client) error {
 				configmap := &corev1.ConfigMap{}
-				if err := c.Get(context.TODO(), newRequest(resourcesNamespace, getSystemProbeConfigConfigMapName(resourcesName)).NamespacedName, configmap); err != nil {
+				if err := c.Get(context.TODO(), newRequest(resourcesNamespace, fmt.Sprintf("%s-%s", resourcesName, SystemProbeConfigMapSuffixName)).NamespacedName, configmap); err != nil {
```
Comment (clamoriniere) [inline controllers/datadogagent/controller_test.go:739]: I think you can still use [getSystemProbeConfigConfigMapName()](https://github.com/DataDog/datadog-operator/blob/8b4b7966c329638b5c7f4fbf9f5a6e6fbc7495c9/controllers/datadogagent/systemprobe.go#L323)

**Review by clamoriniere — COMMENTED** [review]
few small nits, otherwise looks good

File: controllers/datadogagent/controller_test.go (hunk 9 above)
Comment (vboulineau) [reply controllers/datadogagent/controller_test.go:739]: The `dda` object is unfortunately not accessible from this method.

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**